File: java/observer/WeatherRama/observer/displays/CurrentConditionDisplay.java
package observer.displays;

import observer.Display;
import observer.Observer;
import subject.Subject;
import subject.WeatherStation;

public class CurrentConditionDisplay implements Observer, Display {

    private float temperature, humidity, pressure;
    private WeatherStation weatherStation;

    public CurrentConditionDisplay(WeatherStation weatherStation) {
        this.weatherStation = weatherStation;
        weatherStation.registerObserver(this);
    }

    @Override
    public void update(float temperature, float humidity, float pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
        display();
    }

    @Override
    public void display() {
        System.out.println(
                "Current Conditions:\n" +
                        temperature + "F degree\n" +
                        humidity + "% humidity\n" +
                        pressure + " pressure\n"
        );
    }
}

File: kotlin/observer/WeatherRama/observer/displays/CurrentConditionDisplay.kt
package observer.displays

import observer.Display
import observer.Observer
import subject.WeatherStation

class CurrentConditionDisplay(weatherStation: WeatherStation) : Observer, Display {

    private var temperature: Float? = null
    private var humidity: Float? = null
    private var pressure: Float? = null

    private var weatherStation = WeatherStation()

    init {
        this.weatherStation = weatherStation
        weatherStation.registerObserver(this)
    }

    override fun update(temperature: Float, humidity: Float, pressure: Float) {
        this.temperature = temperature
        this.humidity = humidity
        this.pressure = pressure

        display()
    }

    override fun display() {
        println("Current Conditions:\n"
                + temperature + "F degree\n"
                + humidity + "% humidity\n"
                + pressure + " pressure\n"
        )
    }

}